backend/corpora/common/upload_sfn.py

```diff
@@ -17,11 +30,70 @@ def get_stepfunctions_client():
 
 
 def start_upload_sfn(collection_uuid, dataset_uuid, url):
-    input_parameters = {"collection_uuid": collection_uuid, "url": url, "dataset_uuid": dataset_uuid}
+    input_parameters = {
+        "collection_uuid": collection_uuid,
+        "url": url,
+        "dataset_uuid": dataset_uuid,
+    }
     sfn_name = f"{dataset_uuid}_{int(time.time())}"
     response = get_stepfunctions_client().start_execution(
         stateMachineArn=CorporaConfig().upload_sfn_arn,
         name=sfn_name,
         input=json.dumps(input_parameters),
     )
     return response
+
+
+def upload(
+    db_session: Session,
+    collection_uuid: str,
+    user: str,
+    url: str,
+    file_size: int,
+    file_extension: str,
+    dataset_id: str = None,
+    curator_tag: str = None,
+) -> str:
+    max_file_size_gb = CorporaConfig().upload_max_file_size_gb * GB
+    if file_size is not None and file_size > max_file_size_gb:
+        raise MaxFileSizeExceededException(f"{url} exceeds the maximum allowed file size of {max_file_size_gb} Gb")
+
+    allowed_file_formats = CorporaConfig().upload_file_formats
+    if file_extension not in allowed_file_formats:
+        raise InvalidFileFormatException(f"{url} must be in the file format(s): {allowed_file_formats}")
+
+    collection = Collection.get_collection(
+        db_session,
+        collection_uuid,
+        visibility=CollectionVisibility.PRIVATE,  # Do not allow changes to public Collections
+        owner=_owner_or_allowed(user),
```

Comment:
Don't specify `owner`. This is a filter that exists to prevent authenticated APIs to access collections that do not belong to that user.

Reply:
Agreed

Reply:
This is needed, because this code is common to s3 uploads _and_ UX uploads.

Reply:
Yep, we had chatted about this offline.